src/Dapr.AspNetCore/CloudEventsMiddleware.cs

```diff
@@ -83,17 +95,29 @@ private async Task ProcessBodyAsync(HttpContext httpContext, string charSet)
             string contentType;
 
             // Check whether to use data or data_base64 as per https://github.com/cloudevents/spec/blob/v1.0.1/json-format.md#31-handling-of-data
-            var isDataSet = json.TryGetProperty("data", out var data);
-            var isBinaryDataSet = json.TryGetProperty("data_base64", out var binaryData);
+            // Get the property names by OrdinalIgnoreCase comparison to support case insensitive JSON as the Json Serializer for AspCore already supports it by default.
+            var jsonPropNames = json.EnumerateObject().ToArray();
+
+            var dataPropName = jsonPropNames
+                .Select(d => d.Name)
+                .FirstOrDefault(d => d.Equals("data", StringComparison.OrdinalIgnoreCase)) ?? "";
+
+            var dataBase64PropName = jsonPropNames
+                .Select(d => d.Name)
+                .FirstOrDefault(d => d.Equals("data_base64", StringComparison.OrdinalIgnoreCase)) ?? "";
```

Comment:
AFAIK, JSON property names can be empty strings.  Is it really intended to use `""` as a property name should `"data_base64"` (or equivalent) not be found?  Should the empty string just be omitted and allow the property name to be `null`, in which case the following tests could be:

```csharp
var isDataSet = dataPropName is not null && json.TryGetProperty(dataPropName, out var data);
var isBinaryDataSet = dataBase64PropName is not null && json.TryGetProperty(dataBase64PropName, out var binaryData);
```

Reply:
hmm this was under the assumption that the cloud event wont have any empty props but you are correct, there's nothing stopping the payload to have one